packages/styled-components/src/utils/stylis.js

```diff
@@ -0,0 +1,98 @@
+import Stylis from 'stylis/stylis.min';
+import _insertRulePlugin from 'stylis-rule-sheet';
+import { EMPTY_ARRAY, EMPTY_OBJECT } from './empties';
+
+const COMMENT_REGEX = /^\s*\/\/.*$/gm;
+
+export default function createStylisInstance(
+  options: Object = EMPTY_OBJECT,
+  middlewares: Array<Function> = EMPTY_ARRAY
+) {
+  const stylis = new Stylis({
+    global: false,
+    cascade: true,
+    keyframe: false,
+    prefix: true,
+    compress: false,
+    semicolon: false, // NOTE: This means "autocomplete missing semicolons"
+
+    // user-given overrides
+    ...options,
```

Comment:
We don't want users to be able to override the other options, right? E.g. if they set `global: true` or `cascade: false` that's bad, no?

Reply:
Messing with stylis in general is kind of "here be dragons" territory... definitely a pro-level functionality and anyone getting this deep would need to read up on what the options actually do.